main.cpp

```diff
@@ -19,20 +19,33 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
-
+    CastCallBack(Rewriter& rewriter): _rewriter(rewriter) {};
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+        const auto *cast_expression = Result.Nodes.getNodeAs<CStyleCastexpr>("cast");
+        auto replace = CharSourceRange::getCharRange(cast_expression->getLParenLoc(),
+                            cast_expression->getSubexprAsWritten()->getBeginLoc());
+        auto &src_mngmnt = *Result.SourceManager;
+        auto typename = Lexer::getSourceText(CharSourceRange::getTokenRange(
+                                            cast_expression->getLParenLoc().getLocWithOffset(1),
+                                            cast_expression->getRParenLoc().getLocWithOffset(-1)),
+                                            src_mngmnt, LangOptions());
+        const auto *expr = cast_expression->getSubexprAsWritten()->IgnoreImpCasts();

```

Comment:
Please describe why you are using here `getSubExprAsWritten()` and `IgnoreImpCasts()`.
What would be the difference with the use of `getSubExprAsWritten()->IgnoreImpCasts()`, `getSubExpr()->IgnoreImpCasts()` and `getSubExprAsWritten()`?

Reply:
getSubExprAsWritten() gets the cast subexpression as it was written in the source code, looking at any implicit casts or other intermediate nodes introduced by semantic analysis. We then use IgnoreImpCasts() to ignore any implicit casts that may surround this expression.

If we run our code in the test.cpp file, it doesn't matter if we use getSubExprAsWritten() alone or with IgnoreImpCasts(). getSubExprAsWritten()->IgnoreImpCasts(), getSubExpr()->IgnoreImpCasts() and getSubExprAsWritten() will give us the same result.